linalg/densemat.cpp

```diff
@@ -221,8 +221,13 @@ void DenseMatrix::MultTranspose(const Vector &x, Vector &y) const
    MultTranspose((const double *)x, (double *)y);
 }
 
-void DenseMatrix::AddMult(const Vector &x, Vector &y) const
+void DenseMatrix::AddMult(const Vector &x, Vector &y, const double a) const
 {
+   if (a != 1.0)
```

Comment:
Is this check stable? Can't we hit weird roundoff?

Reply:
Also part of #3073, sorry for the confusion. I'm not an expert here, but I was under the impression that this is safe because floating point types can always exactly represent the value 1. There may be a better way to do it Maybe there is a better way to do it.

Reply:
This is stable for the default hard-coded argument `a = 1.0`. Even for round-offs, the method yields the correct results just with some performance overhead.